Add E2E tests for Product Collection with Compatibility Layer

kmanijak opened the pull request:
Recreation of https://github.com/woocommerce/woocommerce-blocks/pull/10463 due to unnnecessarily complex conflicts to resolve.

This PR adds set of E2E Playwright tests for Compatibility Layer against the Product Collection used in Product Archive template.


### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. E2E test are passing

* [x] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

kmanijak committed:
Add E2E tests for Product Collection with Compatibility Layer

kmanijak committed:
Remove unnecessary changes

gigitux commented on tests/e2e/tests/product-collection/compatibility-layer-block_theme.spec.ts:
```diff
@@ -0,0 +1,150 @@
+import {
+	installPluginFromPHPFile,
+	uninstallPluginFromPHPFile,
+} from '../../mocks/custom-plugins';
```
We are trying to importing the utils avoiding the relative import:

```suggestion
} from 'from '@woocommerce/e2e-mocks/custom-plugins';
```

gigitux commented on tests/e2e/tests/product-collection/compatibility-layer-block_theme.spec.ts:
I think that this E2E test contain some side-effects, so we should rename this E2E test with:
compatibility-layer.block_theme.side-effects.spec.ts. Also, currently missing a `.` before block_theme.

The suffix adoption is really important for our E2E test infrastructure.

gigitux commented on tests/e2e/tests/product-collection/compatibility-layer-block_theme.spec.ts:
```diff
@@ -0,0 +1,150 @@
+				);
+			} );
+		}
+	);
```
If I understand well, we are checking that the text is present in the page. We are not checking the order, right?

kmanijak replied:
That's true. I have an idea to test the order but I need a bit more time for it to verify. For now I'd go ahead and merge this PR in its current state. I addressed other comments though.

Thanks for the great review with great tips! 🙇

gigitux commented on tests/e2e/tests/product-collection/compatibility-layer-block_theme.spec.ts:
```diff
@@ -0,0 +1,150 @@
+	);
+
+	test.afterAll( async () => {
+		await uninstallPluginFromPHPFile(
```
Since that we have update the template, we have to cleanup the global change with:
```
	await deleteAllTemplates( 'wp_template' );
```

gigitux reviewed (changes requested):
Thanks for your work! Overall LGTM, but before merging it, we should address some comments!

kmanijak committed:
Rename the file so it covers side effects and improve imports

kmanijak commented:
@gigitux I addressed most of the comments except for:

> If I understand well, we are checking that the text is present in the page. We are not checking the order, right?

> That's true. I have an idea to test the order, but I need more time to verify it. For now I'd go ahead and merge this PR in its current state. I addressed other comments though.

I'd be grateful if you could review again 🙏

gigitux commented on tests/e2e/tests/product-collection/compatibility-layer.block_theme.side-effects.spec.ts:
The right suffix is `side_effects`.

gigitux reviewed (approved):
I’m pre-approving this PR, but before merging the PR it is necessary to update the suffix to `side_effects`.

kmanijak committed:
Rename the file to the correct suffix - side_effects instead of side-effects

kmanijak committed:
Merge branch 'trunk' into add/e2e-tests-compatiblity-layer-2
